features/steps/validation_handling.py

```diff
@@ -164,19 +164,19 @@ def apply_then_operation(fn, inst, context, current_path, depth=0, **kwargs):
 
                 context.gherkin_outcomes.append(validation_outcome)
 
-                if not step_results:
-
-                    validation_outcome = ValidationOutcome(
-                        outcome_code=ValidationOutcomeCode.PASSED,  # todo @gh "Rule passed" # deactivated until code table is added to django model
-                        observed=None,
-                        expected=None,
-                        feature=context.feature.name,
-                        feature_version=misc.define_feature_version(context),
-                        severity=OutcomeSeverity.PASSED,
-                        instance_id = safe_method_call(activation_inst, 'id', None),
-                        validation_task_id=context.validation_task_id
-                    )
-                    context.gherkin_outcomes.append(validation_outcome)
+            if not step_results:
+
+                validation_outcome = ValidationOutcome(
+                    outcome_code=ValidationOutcomeCode.PASSED,  # todo @gh "Rule passed" # deactivated until code table is added to django model
```

Comment:
Is this `todo` still applicable?  To me it looks like all of the outcomes are present in the django model - ref https://github.com/buildingSMART/ifc-validation-data-model/blob/b29efda228a1b019e21b9761fbedbdba0a40f81e/models.py#L979

Reply:
No, not applicable anymore .. 
I actually also don't see any passed results in our database for normative rules or warnings. Which makes sense; the code never gets reached. The reason that the blocks are turned grey is then because 'executed' flags a file as valid.